integration/helper.rb

```diff
@@ -0,0 +1,154 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "fileutils"
+require "open3"
+
+##
+# Keep trying a block of code until the code of block yield a true statement or
+# raise error after timeout
+def keep_trying_till_true timeout = 30
+  t_begin = Time.now
+  loop do
+    if yield
+      break
+    elsif (Time.now - t_begin) > timeout
+      raise RuntimeError, "Timeout after trying for #{timeout} seconds"
```

Comment:
You can just:
```ruby
raise "Timeout after trying for #{timeout} seconds"
```
It automatically picks RuntimeError if you don't specify a class.

Alternately, you can also
```ruby
fail "Timeout after trying for #{timeout} seconds"
```
which does the same thing, and is what you've done in other code below.